### PR TITLE
Blocks: Add Ribbon to indicate Paid blocks in block picker

### PR DESCRIPTION
**Pull request by ockham**
WIP, currently untested.

#### Changes proposed in this Pull Request:

Add a ribbon to visually indicate paid blocks in the block picker

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

It does add a feature  to an existing part of Jetpack :grimacing:

#### Testing instructions:

TBD

#### Proposed changelog entry for your changes:

Blocks: Add Ribbon to indicate Paid blocks in block picker

### CODE REVIEW

File: extensions/shared/register-jetpack-block.js [hunk 1]
```diff
@@ -48,6 +50,14 @@ export default function registerJetpackBlock( name, settings, childBlocks = [] )
 		...settings,
 		title: betaExtensions.includes( name ) ? `${ settings.title } (beta)` : settings.title,
 		edit: requiredPlan ? wrapPaidBlock( { requiredPlan } )( settings.edit ) : settings.edit,
+		icon: requiredPlan ? (
+			<>
```
Comment (ockham) [inline extensions/shared/register-jetpack-block.js]: Need to bump `@automattic/calypso-build` to a more recent version for this to work.

**Comment by ockham** [comment]
I've bumped `calypso-build` to `4.0.1` and added `calypso-ui`. Upon `yarn build`, I'm now getting

```
Error: [BABEL] ~/src/jetpack/gulpfile.babel.js: Cannot find module '@babel/plugin-proposal-class-properties' from '~/src/jetpack'
    at Function.module.exports [as sync] (~/src/jetpack/node_modules/resolve/lib/sync.js:71:15)
    at resolveStandardizedName (~/src/jetpack/node_modules/@babel/core/lib/config/files/plugins.js:101:31)
    at resolvePlugin (~/src/jetpack/node_modules/@babel/core/lib/config/files/plugins.js:54:10)
    at loadPlugin (~/src/jetpack/node_modules/@babel/core/lib/config/files/plugins.js:62:20)
    at createDescriptor (~/src/jetpack/node_modules/@babel/core/lib/config/config-descriptors.js:154:9)
    at items.map (~/src/jetpack/node_modules/@babel/core/lib/config/config-descriptors.js:109:50)
    at Array.map (<anonymous>)
    at createDescriptors (~/src/jetpack/node_modules/@babel/core/lib/config/config-descriptors.js:109:29)
    at createPluginDescriptors (~/src/jetpack/node_modules/@babel/core/lib/config/config-descriptors.js:105:10)
    at plugins (~/src/jetpack/node_modules/@babel/core/lib/config/config-descriptors.js:40:19)
```

Even though that package is a dependency of `calypso-build`.

This seems somewhat similar to p1566905517230900-slack-luna, which we hoped to fix with #13343 :thinking: 

Installing that package directly gives the same kind of error for `@babel/plugin-syntax-dynamic-import`, which is also in `calypso-build`'s dependencies :man_facepalming:

File: package.json [hunk 2]
```diff
@@ -75,8 +75,9 @@
 		}
 	},
 	"dependencies": {
-		"@automattic/calypso-build": "1.0.0",
+		"@automattic/calypso-build": "4.0.1",
```
Comment (simison) [inline package.json]: Updating the build tool is likely easier and safer to do in its own PR?

Comment (sirreal) [reply package.json]: Working on this in https://github.com/Automattic/jetpack/pull/13503

**Comment by ockham** [comment]
Rebased, and bumped `calypso-ui` to `2.0.0-alpha.0`. Getting the following error during `yarn build`:

```
        ERROR in ./node_modules/@automattic/calypso-ui/dist/esm/card/style.scss (./node_modules/css-loader/dist/cjs.js??ref--5-1!./node_modules/postcss-loader/src??ref--5-2!./node_modules/@automattic/calypso-build/node_modules/sass-loader/lib/loader.js??ref--5-3!./node_modules/@automattic/calypso-ui/dist/esm/card/style.scss)
        Module build failed (from ./node_modules/@automattic/calypso-build/node_modules/sass-loader/lib/loader.js):
        
        
                 ^
              No mixin named clear-fix
              in ~/src/jetpack/node_modules/@automattic/calypso-ui/dist/esm/card/style.scss (line 11, column 11)
```

This is because of https://github.com/Automattic/wp-calypso/blob/9ebe145354a7e7ab5540ce2ecd861c61ff7f963b/packages/calypso-ui/src/card/style.scss#L10 which needs https://github.com/Automattic/wp-calypso/blob/9ebe145354a7e7ab5540ce2ecd861c61ff7f963b/client/assets/stylesheets/shared/mixins/_clear-fix.scss.

Ad-hoc solution idea: Bundle mixins in a new package (`calypso-sass-mixins`?) and make them a dep for `calypso-ui`? Thoughts @sirreal @Automattic/team-calypso ?

**Comment by ockham** [comment]
> Some exploratory work was done here, you should be able to use this published package as the sas loader prelude:
> 
> [Automattic/wp-calypso#36170](https://github.com/Automattic/wp-calypso/pull/36170)
> 
> That may fix the issues but isn't really well thought out.

Ah right, had forgotten about that PR.

Right off the bat: We're [injecting](https://github.com/Automattic/wp-calypso/pull/36170/files#diff-11e9f7f953edc64ba14b0cc350ae7b9dR215) the prelude at Webpack config level (through the SASS "preset") there, which is something I'd rather avoid for projects using `@automattic/calypso-ui`. Can we 'simply' change `@automattic/calypso-ui` itself to use the prelude?

**Comment by simison** [comment]
Similar issues highlighted by https://github.com/Automattic/wp-calypso/pull/36471

File: package.json (hunk 2 above)
Comment (sirreal) [reply package.json]: #13503 shipped.

File: extensions/shared/register-jetpack-block.js [hunk 3]
```diff
@@ -49,6 +51,14 @@ export default function registerJetpackBlock( name, settings, childBlocks = [] )
 		title: betaExtensions.includes( name ) ? `${ settings.title } (beta)` : settings.title,
 		edit: requiredPlan ? wrapPaidBlock( { requiredPlan } )( settings.edit ) : settings.edit,
 		example: requiredPlan ? undefined : settings.example,
+		icon: requiredPlan ? (
+			<>
+				<Ribbon>{ __( 'Paid' ) }</Ribbon>
```
Comment (simison) [inline extensions/shared/register-jetpack-block.js]: We should probably give something little more for screen reader users than "paid", like "block name (paid block)".

**Comment by jeherve** [comment]
closing in favor of #14739
